### DISCUSSION TITLE
plaback event button

### DISCUSSION THREAD

**Discussion by robbigoal** [artisan-roaster-scope/artisan#742] [Q&A] [answered]
how do I set the playback event button, my event button settings > type "--" value "1", is there something wrong?, so far the playback event button doesn't work, except the "type" in the slider can only work, THK

**Comment by MAKOMO** [comment, marked as answer]
Not sure what you are talking about. Any more details on what your setup is, what you want to achieve, what  "playback event button" you are talking about and what you mean by "except the type in the slider can only work"?

**Comment by robbigoal** [comment]
sorry my english is not good
I mean like this
>>>>>>> profile background <<<<<<<<<<<<
Time	ET	BT	Description	                  Type	Value
00:02	200.9C	184.8C	Charge Open	    --	1 >>>>>>>> not work following background  (type "--" > not slider)
00:10	172.3C	150.3C	Charge Close	    --	1 >>>>>>>> not work following background  (type "--" > not slider)
00:22	155.5C	116.3C	A1 (S0)		   Air	40 >>>>>>>> work following background  (type Air -- slider)
00:24	154.8C	113.1C	A0 (S1)		  Drum	70  >>>>>>>> work following background (type DRUM -- slider)

>>>>setting button <<<<<
Label	                                     Description	Type	Value	Action	               Documentation	Visibility	Color	Text Color
60	Charge\n Open	                Charge Open	--	       1	  Modbus Command 	wcoil(15,0,1)	ON	#ffff7f	#ffff7f
62	Charge\n Close	                Charge Close	--	       1	 Modbus Command	       wcoil(15,1,1)	ON	#55ffff	#55ffff
15	AirFlow\n90%	                       Rpm - Airflow  Air	        90	  Modbus Command	     write(4,64,{})	        ON	#ff0000	#ff0000
27	Drum\n70%	                      rpm drum	      Drum	        70	   Modbus Command	write(3,9218,{})	ON	#0000ff	#0000ff

so if the type button (--) can't be followed, even though it's recorded, is there something wrong with my settings?

**Reply by MAKOMO** [reply]
Seems your problem is with some custom event buttons that send MODBUS messages out to your connected device. Which device/roasting machine is connected?

Could you please attach your artisan-settings.txt file here (export your settings via menu Help >> Save Settings to a file artisan-settings.aset and rename this one to artisan-settings.txt and attach it here on Github via drag-and-drop).

**Reply by robbigoal** [reply]
thanks for the response and the help

**Comment by robbigoal** [comment]
[artisan-settingsformark - Copy.txt](https://github.com/artisan-roaster-scope/artisan/files/7616975/artisan-settingsformark.-.Copy.txt)

**Comment by MAKOMO** [comment]
Your "PlayBack ON" and "PlayBack OFF" buttons work for me. The first screenshots below is after pressing "PlayBack ON" and the second after pressing "PlayBack OFF".

<img width="1285" alt="Screenshot 2021-11-29 at 11 36 16" src="https://user-images.githubusercontent.com/13909011/143853354-634eafee-821a-415e-b949-1dc6b55e8c58.png">

<img width="1285" alt="Screenshot 2021-11-29 at 11 36 02" src="https://user-images.githubusercontent.com/13909011/143853338-2eedaf48-d443-43f9-a070-db6543eba4ca.png">

Events of type "--" are never replayed. Only events with any of those 4 types that have an associated active slider are replayed.

"Buttons" are never recorded. The replay is by typed events. So this is the expected behavior that you see.

The replay mechanism is an "typed roast event reply" mechanism not a button press replay mechanism.

**Reply by robbigoal** [reply]
thank you for your help and certainty, regarding the type of event "--" will not be replayed,
